shared/src/main/kotlin/com/egm/stellio/shared/web/AuthUtils.kt

```diff
@@ -1,12 +1,14 @@
 package com.egm.stellio.shared.web
 
+import com.egm.stellio.shared.util.toUri
 import org.springframework.security.core.context.ReactiveSecurityContextHolder
 import org.springframework.security.core.context.SecurityContextImpl
 import org.springframework.security.oauth2.jwt.Jwt
 import reactor.core.publisher.Mono
+import java.net.URI
 
-fun extractSubjectOrEmpty(): Mono<String> {
+fun extractSubjectOrEmpty(): Mono<URI> {
     return ReactiveSecurityContextHolder.getContext()
         .switchIfEmpty(Mono.just(SecurityContextImpl()))
-        .map { context -> context.authentication?.principal?.let { (it as Jwt).subject } ?: "" }
+        .map { context -> context.authentication?.principal?.let { ((it as Jwt).subject).toUri() } ?: "".toUri() }
```

Comment:
Hmm, the sub is not an URI

Reply:
But the userId is an URI ?

Reply:
userId (inside Stellio) is an URI, yes. But here, you are extracting the sub, it is not an URI (cf https://openid.net/specs/openid-connect-core-1_0.html#IDToken).

Reply:
So as we did for entityId, we extract the sub as String, then we convert it to URI before being processed in authorizationService ?